### PR TITLE
Fix crash in preload.js due to 3rd-party 'loglevel' (fix #1175, fix #1176)

### LINKED ISSUE #1175: [regression-43.1.0] Injected js broken, with app error "Cannot find module loglevel"
**Homework**

- [ ] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/docs/api.md
- [ ] I checked the [Troubleshooting section of the README](https://github.com/nativefier/nativefier/blob/master/README.md#troubleshooting).
- [ ] I searched existing issues, open & closed. Yes, my bug is new.
- [x] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

Injected js does not work after updating to nativefier 43.1.0. 
This might have something to do with #1162
@TheCleric @ronjouch 

**Steps to reproduce**

```
$ cd /tmp
$ echo 'alert("test");' > injecttest.js
$ nativefier "https://example.com" --inject injecttest.js --verbose
Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
Inferring user agent for electron 12.0.6 / darwin
Grabbing electron<->chrome versions file from https://atom.io/download/atom-shell/index.json
Inferring icon for https://example.com/ on darwin
Inferring icon from store for https://example.com/ on darwin
Allowed icon formats when building for darwin: [ '.icns', '.png', '.ico' ]
Inferring name for https://example.com/
Fetched 1.2 kb page at https://example.com/
Inferred title: Example Domain
Sanitized filename for Example Domain : Example Domain
Got 249 icons from gitcloud
Max icon match score: 0
No relevant icon in store.
Trying to extract a .png icon from the page.
Associated electron v12.0.6 to chrome v89.0.4389.128
Given chrome 89.0.4389.128 on darwin, using user agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.128 Safari/537.36
options {
  packager: {
    appCopyright: undefined,
    appVersion: undefined,
    arch: 'x64',
    asar: false,
    buildVersion: undefined,
    darwinDarkModeSupport: false,
    dir: '/Users/USER/.n/lib/node_modules/nativefier/app',
    electronVersion: '12.0.6',
    icon: undefined,
    name: 'Example Domain',
    out: '/private/tmp',
    overwrite: true,
    platform: 'darwin',
    portable: false,
    targetUrl: 'https://example.com/',
    tmpdir: false,
    upgrade: false,
    upgradeFrom: undefined,
    win32metadata: {
      ProductName: undefined,
      InternalName: undefined,
      FileDescription: undefined
    }
  },
  nativefier: {
    accessibilityPrompt: true,
    alwaysOnTop: false,
    backgroundColor: null,
    basicAuthPassword: null,
    basicAuthUsername: null,
    blockExternalUrls: false,
    bookmarksMenu: null,
    bounce: false,
    browserwindowOptions: undefined,
    clearCache: false,
    counter: false,
    crashReporter: undefined,
    disableContextMenu: undefined,
    disableDevTools: undefined,
    disableGpu: false,
    diskCacheSize: null,
    disableOldBuildWarning: false,
    enableEs3Apis: false,
    fastQuit: false,
    fileDownloadOptions: undefined,
    flashPluginDir: null,
    fullScreen: false,
    globalShortcuts: null,
    hideWindowFrame: undefined,
    ignoreCertificate: false,
    ignoreGpuBlacklist: false,
    inject: [ 'injecttest.js' ],
    insecure: false,
    internalUrls: null,
    lang: undefined,
    maximize: false,
    nativefierVersion: '43.1.0',
    processEnvs: undefined,
    proxyRules: null,
    showMenuBar: false,
    singleInstance: false,
    titleBarStyle: null,
    tray: false,
    userAgent: 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_12_5) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.128 Safari/537.36',
    userAgentOverriden: false,
    verbose: true,
    versionString: undefined,
    width: 1280,
    height: 800,
    minWidth: undefined,
    minHeight: undefined,
    maxWidth: undefined,
    maxHeight: undefined,
    widevine: false,
    x: undefined,
    y: undefined,
    zoom: 1
  }
}

Preparing Electron app...
Copying electron app from /Users/USER/.n/lib/node_modules/nativefier/app to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG
Writing app config to /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG/nativefier.json
Copying 1 files to inject in app.
Copying injection file "injecttest.js" to "/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG/inject/inject-34fd6a..js"
Updating /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG/package.json 'name' field to example-domain-nativefier-944894

Converting icons...
Option "icon" not set, skipping icon conversion.
Copying icons if necessary
No icon specified in options; aborting

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v16.0.0
  electron-packager Host Operating system: darwin 17.7.0 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":false,"dir":"/var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG","electronVersion":"12.0.6","name":"Example Domain","out":"/private/tmp","overwrite":true,"platform":"darwin","portable":false,"targetUrl":"https://example.com/","tmpdir":false,"upgrade":false,"win32metadata":{}} +0ms
  electron-packager Target Platforms: darwin +1ms
  electron-packager Target Architectures: x64 +0ms
  electron-packager Inferring appVersion from version in /var/folders/x4/d6xhrn615_15s2bdvqp7rtw80000gr/T/nativefier-10-50-47-app--80069-HCGnxGhTGTLG/package.json +0ms
  electron-packager Application name: Example Domain +9ms
  electron-packager Target Electron version: 12.0.6 +0ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"darwin","arch":"x64","version":"12.0.6","artifactName":"electron"} +0ms
Packaging app for platform darwin x64 using electron v12.0.6
  electron-packager Creating /private/tmp/darwin-x64-template +4ms
  electron-packager Extracting /Users/USER/Library/Caches/electron/739809e58a9d0011b22ab70d11516a49a099dadf81ec40aabf349af3c1154da6/electron-v12.0.6-darwin-x64.zip to /private/tmp/darwin-x64-template +0ms
  electron-packager Initializing app in /private/tmp/Example Domain-darwin-x64 from /private/tmp/darwin-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  '/private/tmp/Example Domain-darwin-x64',
  '/private/tmp/Example Domain-darwin-arm64',
  '/private/tmp/Example Domain-linux-ia32',
  '/private/tmp/Example Domain-linux-x64',
  '/private/tmp/Example Domain-linux-armv7l',
  '/private/tmp/Example Domain-linux-arm64',
  '/private/tmp/Example Domain-linux-mips64el',
  '/private/tmp/Example Domain-mas-x64',
  '/private/tmp/Example Domain-mas-arm64',
  '/private/tmp/Example Domain-win32-ia32',
  '/private/tmp/Example Domain-win32-x64',
  '/private/tmp/Example Domain-win32-arm64'
] +10s
  electron-packager Renaming Electron to Example Domain in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/MacOS +160ms
  electron-packager Renaming Electron Helper to Example Domain Helper in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper.app/Contents/MacOS +1ms
  electron-packager Renaming Electron Helper (Renderer) to Example Domain Helper (Renderer) in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Renderer).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (Plugin) to Example Domain Helper (Plugin) in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (Plugin).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper (GPU) to Example Domain Helper (GPU) in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks/Electron Helper (GPU).app/Contents/MacOS +0ms
  electron-packager Renaming Electron Helper.app to Example Domain Helper.app in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +1ms
  electron-packager Renaming Electron Helper (Renderer).app to Example Domain Helper (Renderer).app in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (Plugin).app to Example Domain Helper (Plugin).app in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms
  electron-packager Renaming Electron Helper (GPU).app to Example Domain Helper (GPU).app in /private/tmp/Example Domain-darwin-x64/Electron.app/Contents/Frameworks +0ms

Finalizing build...
App built to /private/tmp/Example Domain-darwin-x64, move to wherever it makes sense for you and run the app bundle.
```

**Expected behavior**
nativefier@43.0.2
Screen Shot 2021-05-01 at 10 57 06 AM

**Actual behavior**
nativefier@43.1.0
Screen Shot 2021-05-01 at 10 51 53 AM

**Debug info**

- Console logs of your `nativefier` build command, with `--verbose` flag
- If the bug happens at app run time, the in-app DevTools console logs (open it with F12)
- Error messages
- Screenshots
- Anything else relevant!

**Context**

 - Nativefier: 43.1.0
 - Node.js: v16.0.0
 - Npm: 7.10.0
 - OS: macOS 10.13.6
 - Is it a regression? Yes, nativefier@43.0.2 worked
 - Additional context: (for example: "I'm behind a proxy, with configuration X and protocol Y")

### LINKED ISSUE #1176: [regression-43.1.0] App won't stay logged in after update to v43.1.0
**Homework**

- [x] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/docs/api.md
- [x] I checked the [Troubleshooting section of the README](https://github.com/nativefier/nativefier/blob/master/README.md#troubleshooting).
- [x] I searched existing issues, open & closed. Yes, my bug is new.
- [x] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Bug description**

I use nativefier to create an application out of my Home Assistant webpage. This worked very well for the ~ last two years.
Today I updated nativefier to version v43.1.0 by running `npm install -g nativefier`. After that I recreated my two applications with the following command
```
nativefier --name "My Application" --icon "D:\Nativefier\Home_Assistant_Logo.ico" --maximize "https://my.domain.com"
```
After launching the applications, I was logged out of my instance. That never happend before after an update. I logged in again, but on reloading the page or closing the app and reopen it, I was logged out again. This happens every single time. 

When I downgrade to v43.0.2 via `npm install -g nativefier@43.0.2` everything works fine again. I'm even logged in after recompiling and opening the application (without any login). 

**Steps to reproduce**

Give your ***full*** nativefier command and its logs, with the ***`--verbose` flag***, on a ***public*** site:

```
nativefier --verbose --name "Google" --icon "D:\Nativefier\Home_Assistant_Logo.ico" --maximize "https://google.com"

PS D:\Nativefier> nativefier --verbose --name "Google" --icon "D:\Nativefier\Home_Assistant_Logo.ico" --maximize "https://google.com"
Running in verbose mode! This will produce a mountain of logs and is recommended only for troubleshooting or if you like Shakespeare.

Performing async options post-processing.
Inferring user agent for electron 12.0.6 / win32
Grabbing electron<->chrome versions file from https://atom.io/download/atom-shell/index.json
Got icon from options. Using it, no inferring needed
Got name Google from options. No inferring needed
Sanitized filename for Google : Google
Associated electron v12.0.6 to chrome v89.0.4389.128
Given chrome 89.0.4389.128 on win32, using user agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.128 Safari/537.36options {
  packager: {
    appCopyright: undefined,
    appVersion: undefined,
    arch: 'x64',
    asar: false,
    buildVersion: undefined,
    darwinDarkModeSupport: false,
    dir: 'C:\\Users\\User\\scoop\\persist\\nvm\\nodejs\\v12.22.1\\node_modules\\nativefier\\app',
    electronVersion: '12.0.6',
    icon: 'D:\\Nativefier\\Home_Assistant_Logo.ico',
    name: 'Google',
    out: 'D:\\Dev\\HomeAutomation\\Nativefier',
    overwrite: true,
    platform: 'win32',
    portable: false,
    targetUrl: 'https://google.com/',
    tmpdir: false,
    upgrade: false,
    upgradeFrom: undefined,
    win32metadata: {
      ProductName: 'Google',
      InternalName: 'Google',
      FileDescription: 'Google'
    }
  },
  nativefier: {
    accessibilityPrompt: true,
    alwaysOnTop: false,
    backgroundColor: null,
    basicAuthPassword: null,
    basicAuthUsername: null,
    blockExternalUrls: false,
    bookmarksMenu: null,
    bounce: false,
    browserwindowOptions: undefined,
    clearCache: false,
    counter: false,
    crashReporter: undefined,
    disableContextMenu: undefined,
    disableDevTools: undefined,
    disableGpu: false,
    diskCacheSize: null,
    disableOldBuildWarning: false,
    enableEs3Apis: false,
    fastQuit: false,
    fileDownloadOptions: undefined,
    flashPluginDir: null,
    fullScreen: false,
    globalShortcuts: null,
    hideWindowFrame: undefined,
    ignoreCertificate: false,
    ignoreGpuBlacklist: false,
    inject: [],
    insecure: false,
    internalUrls: null,
    lang: undefined,
    maximize: true,
    nativefierVersion: '43.1.0',
    processEnvs: undefined,
    proxyRules: null,
    showMenuBar: false,
    singleInstance: false,
    titleBarStyle: null,
    tray: false,
    userAgent: 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.128 Safari/537.36',
    userAgentOverriden: false,
    verbose: true,
    versionString: undefined,
    width: 1280,
    height: 800,
    minWidth: undefined,
    minHeight: undefined,
    maxWidth: undefined,
    maxHeight: undefined,
    widevine: false,
    x: undefined,
    y: undefined,
    zoom: 1
  }
}

Preparing Electron app...
Copying electron app from C:\Users\User\scoop\persist\nvm\nodejs\v12.22.1\node_modules\nativefier\app to C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4
Writing app config to C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4\nativefier.json
No files to inject, skipping copy.
Updating C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4\package.json 'name' field to google-nativefier-236660

Converting icons...
Building for Windows and icon is already a .ico, no conversion needed
Copying icons if necessary
Copying icon D:\Nativefier\Home_Assistant_Logo.ico to C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4\icon.ico

Packaging... This will take a few seconds, maybe minutes if the requested Electron isn't cached yet...
  electron-packager Electron Packager 15.2.0
  electron-packager Node v12.22.1
  electron-packager Host Operating system: win32 10.0.19043 (x64) +0ms
  electron-packager Packager Options: {"arch":"x64","asar":false,"darwinDarkModeSupport":false,"dir":"C:\\Users\\User\\AppData\\Local\\Temp\\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4","electronVersion":"12.0.6","icon":"D:\\Nativefier\\Home_Assistant_Logo.ico","name":"Google","out":"D:\\Dev\\HomeAutomation\\Nativefier","overwrite":true,"platform":"win32","portable":false,"targetUrl":"https://google.com/","tmpdir":false,"upgrade":false,"win32metadata":{"ProductName":"Google","InternalName":"Google","FileDescription":"Google"}} +1ms
  electron-packager Target Platforms: win32 +1ms
  electron-packager Target Architectures: x64 +1ms
  electron-packager Requiring author in package.json, as CompanyName was not specified for win32metadata +0ms
  electron-packager Inferring appVersion from version in C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4\package.json +20ms
  electron-packager Inferring win32metadata.CompanyName from author in C:\Users\User\AppData\Local\Temp\nativefier-1-24-48-app--20824-HhZ8NZAIFGP4\package.json +1ms
  electron-packager Application name: Google +22ms
  electron-packager Target Electron version: 12.0.6 +1ms
  electron-packager Ignored path regular expressions: [
  '/package-lock\\.json$',
  '/yarn\\.lock$',
  '/\\.git($|/)',
  '/node_modules/\\.bin($|/)',
  '\\.o(bj)?$'
] +0ms
  electron-packager Downloading Electron with options {"platform":"win32","arch":"x64","version":"12.0.6","artifactName":"electron"} +0ms
Packaging app for platform win32 x64 using electron v12.0.6
  electron-packager Creating D:\Nativefier\win32-x64-template +7ms
  electron-packager Extracting C:\Users\User\AppData\Local\electron\Cache\739809e58a9d0011b22ab70d11516a49a099dadf81ec40aabf349af3c1154da6\electron-v12.0.6-win32-x64.zip to D:\Nativefier\win32-x64-template +2ms
  electron-packager Initializing app in D:\Nativefier\Google-win32-x64 from D:\Nativefier\win32-x64-template template +0ms
  electron-packager Ignored paths based on the out param: [
  'D:\\Nativefier\\Google-darwin-x64',
  'D:\\Nativefier\\Google-darwin-arm64',
  'D:\\Nativefier\\Google-linux-ia32',
  'D:\\Nativefier\\Google-linux-x64',
  'D:\\Nativefier\\Google-linux-armv7l',
  'D:\\Nativefier\\Google-linux-arm64',
  'D:\\Nativefier\\Google-linux-mips64el',
  'D:\\Nativefier\\Google-mas-x64',
  'D:\\Nativefier\\Google-mas-arm64',
  'D:\\Nativefier\\Google-win32-ia32',
  'D:\\Nativefier\\Google-win32-x64',
  'D:\\Nativefier\\Google-win32-arm64'
] +2s
  electron-packager Renaming electron.exe to Google.exe in D:\Nativefier\Google-win32-x64 +47ms
  electron-packager Running rcedit with the options {"version-string":{"FileDescription":"Google","InternalName":"Google","OriginalFilename":"Google.exe","ProductName":"Google","CompanyName":"Jia Hao"},"file-version":"1.0.0","product-version":"1.0.0","icon":"D:\\Nativefier\\Home_Assistant_Logo.ico"} +0ms

Finalizing build...
App built to D:\Nativefier\Google-win32-x64, move to wherever it makes sense for you and run the contained .exe file.
```

**Expected behavior**

Stay logged in like in the versions before.

**Actual behavior**

I don't stay logged in with the current version.

**Debug info**

none

**Context**

 - Nativefier: 43.1.0
 - Node.js: 14.8.0
 - Npm: 6.14.7
 - OS: Windows 10 build 21H1
 - Is it a regression? If yes, what's the last working / first broken version? 
Last working: 43.0.2 
First broken: 43.1.0
 - Additional context: none

### PR DESCRIPTION
**Pull request by TheCleric**
Should fix #1175 and #1176

### CODE REVIEW

**Comment by ronjouch** [comment]
Hey @TheCleric. I agree with the intention to short-term revert to get back to not being broken, but I wonder if commenting only in `app/src/preload.ts` is enough. I'm making a few experiments on my side, coming back here asap with my findings.

**Comment by TheCleric** [comment]
> Hey @TheCleric. I agree with the intention to short-term revert to get back to not being broken, but I wonder if commenting only in `app/src/preload.ts` is enough. I'm making a few experiments on my side, coming back here asap with my findings.

I installed nativefier from npm, and fixed the preload.js in it with the equivalent of this change, and that seemed sufficient. More tests would always be welcome though.

File: .eslintrc.js [hunk 1]
```diff
@@ -16,7 +16,11 @@ module.exports = {
   ],
   rules: {
     'no-console': 'error',
-    'prettier/prettier': 'error',
+    'prettier/prettier': [
+      'error', {
+        'endOfLine': 'auto'
+      }
+    ],
```
Comment (ronjouch) [inline .eslintrc.js:23]: 👍

**Review by ronjouch — COMMENTED** [review]
> I installed nativefier from npm, and fixed the preload.js in it with the equivalent of this change, and that seemed sufficient. More tests would always be welcome though.

@TheCleric I confirm this, I confirm that what you did is enough (loglevel calls elsewhere are unproblematic) and added some doc to your PR. You'll see from my comment that I'm not too sure if *we* are doing something bad with `preload`, or if that's expected behavior, and the doc isn't explicit about it either.

Whatevs. Thanks for the patch, merging and cutting a patch release.

**Pull request merged**